setup.py

```diff
@@ -1,60 +0,0 @@
-"""A setuptools based setup module for dowhy.
-
-Adapted from:
-https://packaging.python.org/en/latest/distributing.html
-https://github.com/pypa/sampleproject
-"""
-
-from setuptools import setup, find_packages
-from codecs import open
-from os import path
-
-here = path.abspath(path.dirname(__file__))
-
-# Get the long description from the README file
-with open(path.join(here, 'README.rst'), encoding='utf-8') as f:
-    long_description = f.read()
-
-# Get the required packages
-with open(path.join(here, 'requirements.txt'), encoding='utf-8') as f:
-    install_requires = f.read().splitlines()
-
-# Plotting packages are optional to install
-extras = ["plotting"]
-extras_require = dict()
-for e in extras:
-    req_file = "requirements-{0}.txt".format(e)
-    with open(req_file) as f:
-        extras_require[e] = [line.strip() for line in f]
-
-# Loading version number
-with open(path.join(here, 'dowhy', 'VERSION')) as version_file:
-    version = version_file.read().strip()
-    print(version)
-
-setup(
```

Comment:
are all of these setup parameters captured in poetry? I'm thinking about `long_description`, `packages`, and some of the keywords below.

Reply:
I encourage you guys verify this for yourself by running `poetry build` and inspecting the tar.gz file in `dist/`, but the long_description, license, keywords, and classifiers are all populated in a generated `setup.py` file. 

long_description is populated by setting the `readme` property in pyproject.toml

Reply:
setup.py: 

```py
# -*- coding: utf-8 -*-
from setuptools import setup

packages = \
['dowhy',
 'dowhy.api',
 'dowhy.causal_estimators',
 'dowhy.causal_identifiers',
 'dowhy.causal_refuters',
 'dowhy.data_transformers',
 'dowhy.do_samplers',
 'dowhy.gcm',
 'dowhy.gcm.independence_test',
 'dowhy.gcm.ml',
 'dowhy.gcm.util',
 'dowhy.graph_learners',
 'dowhy.interpreters',
 'dowhy.utils',
 'dowhy.utils.dgps']

package_data = \
{'': ['*']}

install_requires = \
['joblib>=1.1.0,<2.0.0',
 'llvmlite>=0.31.0,<0.32.0',
 'networkx>=2.8.5,<3.0.0',
 'numpy>=1.23.1,<2.0.0',
 'pandas>=1.4.3,<2.0.0',
 'pydot>=1.4.2,<2.0.0',
 'scikit-learn==1.0.2',
 'scipy>=1.8.1,<2.0.0',
 'statsmodels>=0.13.2,<0.14.0',
 'sympy>=1.10.1,<2.0.0',
 'tqdm>=4.64.0,<5.0.0']

extras_require = \
{'plotting': ['matplotlib>=3.5.2,<4.0.0']}

setup_kwargs = {
    'name': 'dowhy',
    'version': '0.8.0',
    'description': 'DoWhy is a Python library for causal inference that supports explicit modeling and testing of causal assumptions',
    'long_description': '|BuildStatus|_ |PyPiVersion|_ |PythonSupport|_ |Downloads|_\n\n.. |PyPiVersion| image:: https://img.shields.io/pypi/v/dowhy.svg\n.. _PyPiVersion: https://pypi.org/project/dowhy/\n\n.. |PythonSupport| image:: https://img.shields.io/pypi/pyversions/dowhy.svg\n.. _PythonSupport: https://pypi.org/project/dowhy/\n\n.. |BuildStatus| image:: https://github.com/microsoft/dowhy/workflows/Python%20package/badge.svg\n.. _BuildStatus: https://github.com/microsoft/dowhy/actions\n\n.. |Downloads| image:: https://pepy.tech/badge/dowhy\n.. _Downloads: https://pepy.tech/project/dowhy\n\nDoWhy | An end-to-end library for causal inference\n===================================================\n\n  Introducing DoWhy and the 4 steps of causal inference | `Microsoft Research Blog <https://www.microsoft.com/en-us/research/blog/dowhy-a-library-for-causal-inference/>`_ | `Video Tutorial <https://note.microsoft.com/MSR-Webinar-DoWhy-Library-Registration-On-Demand.html>`_ | `Arxiv Paper <https://arxiv.org/abs/2011.04216>`_ | `Slides <https://www2.slideshare.net/AmitSharma315/dowhy-an-endtoend-library-for-causal-inference>`_\n\n  Read the `docs <https://py-why.github.io/dowhy/>`_ | Try it online! |Binder|_ \n\n.. |Binder| image:: https://mybinder.org/badge_logo.svg\n.. _Binder: https://mybinder.org/v2/gh/microsoft/dowhy/master?filepath=docs%2Fsource%2F\n\n**Case Studies using DoWhy**: `Hotel booking cancellations <https://towardsdatascience.com/beyond-predictive-models-the-causal-story-behind-hotel-booking-cancellations-d29e8558cbaf>`_ | `Effect of customer loyalty programs <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_example_effect_of_memberrewards_program.ipynb>`_ | `Optimizing article headlines <https://medium.com/@akelleh/introducing-the-do-sampler-for-causal-inference-a3296ea9e78d>`_ | `Effect of home visits on infant health (IHDP) <https://towardsdatascience.com/implementing-causal-inference-a-key-step-towards-agi-de2cde8ea599>`_ | `Causes of customer churn/attrition <https://medium.com/geekculture/a-quickstart-for-causal-analysis-decision-making-with-dowhy-2ce2d4d1efa9>`_\n\n.. image:: https://raw.githubusercontent.com/microsoft/dowhy/master/docs/images/dowhy-schematic.png\n\nAs computing systems are more frequently and more actively intervening in societally critical domains such as healthcare, education, and governance, it is critical to correctly predict and understand the causal effects of these interventions. Without an A/B test, conventional machine learning methods, built on pattern recognition and correlational analyses, are insufficient for decision-making. \n\nMuch like machine learning libraries have done for prediction, **"DoWhy" is a Python library that aims to spark causal thinking and analysis**. DoWhy provides a principled four-step interface for causal inference that focuses on explicitly modeling causal assumptions and validating them as much as possible. The key feature of DoWhy is its state-of-the-art refutation API that can automatically test causal assumptions for any estimation method, thus making inference more robust and accessible to non-experts. DoWhy supports estimation of the average causal effect for backdoor, frontdoor, instrumental variable and other identification methods, and estimation of the conditional effect (CATE) through an integration with the EconML library.  \n\nFor a quick introduction to causal inference, check out `amit-sharma/causal-inference-tutorial <https://github.com/amit-sharma/causal-inference-tutorial/>`_. We also gave a more comprehensive tutorial at the ACM Knowledge Discovery and Data Mining (`KDD 2018 <http://www.kdd.org/kdd2018/>`_) conference: `causalinference.gitlab.io/kdd-tutorial <http://causalinference.gitlab.io/kdd-tutorial/>`_. For an introduction to the four steps of causal inference and its implications for machine learning, you can access this video tutorial from Microsoft Research: `DoWhy Webinar <https://note.microsoft.com/MSR-Webinar-DoWhy-Library-Registration-On-Demand.html>`_.\n\nDocumentation for DoWhy is available at `py-why.github.io/dowhy <https://py-why.github.io/dowhy/>`_.\n\n.. i here comment toctree::\n.. i here comment   :maxdepth: 4\n.. i here comment   :caption: Contents:\n.. contents:: **Contents**\n\nNews\n-----\n**2022.05.27**:\n\n* **DoWhy now part of PyWhy**\n\n  We have moved DoWhy from microsoft/dowhy to py-why/dowhy. While GitHub will automatically\n  redirect your git command for cloning, pulling, etc., we recommend updating git remotes and bookmarks. Please note\n  that the **documentation** has now moved to https://py-why.github.io/dowhy with **no** redirect from the old URL.\n\n* **Experimental support for GCM-based inference**\n\n  We have started adding support for graphical causal model-based inference (or in short GCM-based). At the moment,\n  this includes support for interventions, counterfactuals, and attributing distribution changes. As part of this,\n  we also added features for Shapley value estimation and independence tests. We\'re still in the process of fleshing\n  everything out, including `documentation <https://py-why.github.io/dowhy/gcm/>`_. Some of it is already on `master\n  <https://github.com/py-why/dowhy>`_, other parts are on feature branches (prefixed with ``gcm-``) with open\n  pull-requests, other parts will appear as new pull-requests in the next couple of weeks. Be sure to watch this space\n  here as we quickly expand functionality and documentation.\n\nThe need for causal inference\n----------------------------------\n\nPredictive models uncover patterns that connect the inputs and outcome in observed data. To intervene, however, we need to estimate the effect of changing an input from its current value, for which no data exists. Such questions, involving estimating a *counterfactual*, are common in decision-making scenarios.\n\n* Will it work?\n    * Does a proposed change to a system improve people\'s outcomes?\n* Why did it work?\n    * What led to a change in a system\'s outcome?\n* What should we do?\n    * What changes to a system are likely to improve outcomes for people?\n* What are the overall effects?\n    * How does the system interact with human behavior?\n    * What is the effect of a system\'s recommendations on people\'s activity?\n\nAnswering these questions requires causal reasoning. While many methods exist\nfor causal inference, it is hard to compare their assumptions and robustness of results. DoWhy makes three contributions,\n\n1. Provides a principled way of modeling a given problem as a causal graph so\n   that all assumptions are explicit.\n2. Provides a unified interface for many popular causal inference methods, combining the two major frameworks of graphical models and potential outcomes.\n3. Automatically tests for the validity of assumptions if possible and assesses\n   the robustness of the estimate to violations.\n\nTo see DoWhy in action, check out how it can be applied to estimate the effect\nof a subscription or rewards program for customers [`Rewards notebook\n<https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_example_effect_of_memberrewards_program.ipynb>`_] and for implementing and evaluating causal inference methods on benchmark datasets like the `Infant Health and Development Program (IHDP) <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_ihdp_data_example.ipynb>`_ dataset, `Infant Mortality (Twins) <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_twins_example.ipynb>`_ dataset, and the `Lalonde Jobs <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_lalonde_example.ipynb>`_ dataset.\n\n\nInstallation\n-------------\n\nDoWhy support Python 3.6+. To install, you can use pip or conda. \n\n**Latest Release**\n\nInstall the latest `release <https://pypi.org/project/dowhy/>`__ using pip.\n\n.. code:: shell\n   \n   pip install dowhy\n   \nInstall the latest `release <https://anaconda.org/conda-forge/dowhy>`__ using conda.\n\n.. code:: shell\n\n   conda install -c conda-forge dowhy\n\nIf you face "Solving environment" problems with conda, then try :code:`conda update --all` and then install dowhy. If that does not work, then use :code:`conda config --set channel_priority false` and try to install again. If the problem persists, please add your issue `here <https://github.com/microsoft/dowhy/issues/197>`_.\n\n**Development Version**\n\nIf you prefer the latest dev version, clone this repository and run the following command from the top-most folder of\nthe repository.\n\n.. code:: shell\n    \n    pip install -e .\n\n**Requirements**\n\nDoWhy requires the following packages:\n\n* numpy\n* scipy\n* scikit-learn\n* pandas\n* networkx  (for analyzing causal graphs)\n* matplotlib (for general plotting)\n* sympy (for rendering symbolic expressions)\n\nIf you face any problems, try installing dependencies manually.\n\n.. code:: shell\n    \n    pip install -r requirements.txt\n\nOptionally, if you wish to input graphs in the dot format, then install pydot (or pygraphviz).\n\n\nFor better-looking graphs, you can optionally install pygraphviz. To proceed,\nfirst install graphviz and then pygraphviz (on Ubuntu and Ubuntu WSL).\n\n.. code:: shell\n\n    sudo apt install graphviz libgraphviz-dev graphviz-dev pkg-config\n    ## from https://github.com/pygraphviz/pygraphviz/issues/71\n    pip install pygraphviz --install-option="--include-path=/usr/include/graphviz" \\\n    --install-option="--library-path=/usr/lib/graphviz/"\n\nSample causal inference analysis in DoWhy\n-------------------------------------------\nMost DoWhy\nanalyses for causal inference take 4 lines to write, assuming a\npandas dataframe df that contains the data:\n\n.. code:: python\n\n    from dowhy import CausalModel\n    import dowhy.datasets\n\n    # Load some sample data\n    data = dowhy.datasets.linear_dataset(\n        beta=10,\n        num_common_causes=5,\n        num_instruments=2,\n        num_samples=10000,\n        treatment_is_binary=True)\n\nDoWhy supports two formats for providing the causal graph: `gml <https://github.com/GunterMueller/UNI_PASSAU_FMI_Graph_Drawing>`_ (preferred) and `dot <http://www.graphviz.org/documentation/>`_. After loading in the data, we use the four main operations in DoWhy: *model*,\n*estimate*, *identify* and *refute*:\n\n.. code:: python\n\n    # I. Create a causal model from the data and given graph.\n    model = CausalModel(\n        data=data["df"],\n        treatment=data["treatment_name"],\n        outcome=data["outcome_name"],\n        graph=data["gml_graph"])\n\n    # II. Identify causal effect and return target estimands\n    identified_estimand = model.identify_effect()\n\n    # III. Estimate the target estimand using a statistical method.\n    estimate = model.estimate_effect(identified_estimand,\n                                     method_name="backdoor.propensity_score_matching")\n\n    # IV. Refute the obtained estimate using multiple robustness checks.\n    refute_results = model.refute_estimate(identified_estimand, estimate,\n                                           method_name="random_common_cause")\n\nDoWhy stresses on the interpretability of its output. At any point in the analysis,\nyou can inspect the untested assumptions, identified estimands (if any) and the\nestimate (if any). Here\'s a sample output of the linear regression estimator.\n\n.. image:: https://raw.githubusercontent.com/microsoft/dowhy/master/docs/images/regression_output.png\n\nFor a full code example, check out the `Getting Started with DoWhy <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_simple_example.ipynb>`_ notebook. You can also use Conditional Average Treatment Effect (CATE) estimation methods from other libraries such as EconML and CausalML, as shown in the `Conditional Treatment Effects <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy-conditional-treatment-effects.ipynb>`_ notebook. For more examples of using DoWhy, check out the Jupyter notebooks in `docs/source/example_notebooks <https://github.com/microsoft/dowhy/tree/master/docs/source/example_notebooks/>`_ or try them online at `Binder <https://mybinder.org/v2/gh/microsoft/dowhy/master?filepath=docs%2Fsource%2F>`_. \n\n\nGCM-based inference (experimental)\n----------------------------------\n\nGraphical causal model-based inference, or GCM-based inference for short, is an experimental addition to DoWhy. For\ndetails, check out the `documentation for the gcm sub-package <https://py-why.github.io/dowhy/gcm>`_. The basic\nrecipe for this API works as follows:\n\n.. code:: python\n\n    # 1. Modeling cause-effect relationships as a structural causal model\n    #    (causal graph + functional causal models):\n    scm = gcm.StructuralCausalModel(nx.DiGraph([(\'X\', \'Y\'), (\'Y\', \'Z\')])) # X -> Y -> Z\n    scm.set_causal_mechanism(\'X\', gcm.EmpiricalDistribution())\n    scm.set_causal_mechanism(\'Y\', gcm.AdditiveNoiseModel(gcm.ml.create_linear_regressor()))\n    scm.set_causal_mechanism(\'Z\', gcm.AdditiveNoiseModel(gcm.ml.create_linear_regressor()))\n\n    # 2. Fitting the SCM to the data:\n    gcm.fit(scm, data)\n\n    # 3. Answering a causal query based on the SCM:\n    results = gcm.<causal_query>(scm, ...)\n\nWhere <causal_query> can be one of multiple functions explained in `Answering Causal Questions <https://py-why.github.io/dowhy/gcm/user_guide/answering_causal_questions/index.html>`_.\n\n\nA high-level Pandas API\n-----------------------\n\nWe\'ve made an even simpler API for dowhy which is a light layer on top of the standard one. The goal is to make causal analysis much more like regular exploratory analysis. To use this API, simply\nimport :code:`dowhy.api`. This will magically add the :code:`causal` namespace to your\n:code:`pandas.DataFrame` s. Then,\nyou can use the namespace as follows.\n\n.. code:: python\n\n    import dowhy.api\n    import dowhy.datasets\n\n    data = dowhy.datasets.linear_dataset(beta=5,\n        num_common_causes=1,\n        num_instruments = 0,\n        num_samples=1000,\n        treatment_is_binary=True)\n\n    # data[\'df\'] is just a regular pandas.DataFrame\n    data[\'df\'].causal.do(x=\'v0\', # name of treatment variable\n                         variable_types={\'v0\': \'b\', \'y\': \'c\', \'W0\': \'c\'},\n                         outcome=\'y\',\n                         common_causes=[\'W0\']).groupby(\'v0\').mean().plot(y=\'y\', kind=\'bar\')\n\n.. image:: https://raw.githubusercontent.com/microsoft/dowhy/master/docs/images/do_barplot.png\n\nFor some methods, the :code:`variable_types` field must be specified. It should be a :code:`dict`, where the keys are\nvariable names, and values are \'o\' for ordered discrete, \'u\' for un-ordered discrete, \'d\' for discrete, or \'c\'\nfor continuous.\n\n**Note:If the** :code:`variable_types` **is not specified we make use of the following implicit conversions:**\n::\n\n   int -> \'c\'\n   float -> \'c\'\n   binary -> \'b\'\n   category -> \'d\'\n\n**Currently we have not added support for timestamps.**\n\nThe :code:`do` method in the causal namespace generates a random sample from $P(outcome|do(X=x))$ of the\nsame length as your data set, and returns this outcome as a new :code:`DataFrame`. You can continue to perform\nthe usual :code:`DataFrame` operations with this sample, and so you can compute statistics and create plots\nfor causal outcomes!\n\nThe :code:`do` method is built on top of the lower-level :code:`dowhy` objects, so can still take a graph and perform\nidentification automatically when you provide a graph instead of :code:`common_causes`.\n\nFor more details, check out the `Pandas API\n<https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_causal_api.ipynb>`_ notebook or the `Do Sampler <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/do_sampler_demo.ipynb>`_\nnotebook. \n\nGraphical Models and Potential Outcomes: Best of both worlds\n============================================================\nDoWhy builds on two of the most powerful frameworks for causal inference:\ngraphical models and potential outcomes. It uses graph-based criteria and\ndo-calculus for modeling assumptions and identifying a non-parametric causal effect.\nFor estimation, it switches to methods based primarily on potential outcomes.\n\nA unifying language for causal inference\n----------------------------------------\n\nDoWhy is based on a simple unifying language for causal inference. Causal\ninference may seem tricky, but almost all methods follow four key steps:\n\n1. Model a causal inference problem using assumptions.\n2. Identify an expression for the causal effect under these assumptions ("causal estimand").\n3. Estimate the expression using statistical methods such as matching or instrumental variables.\n4. Finally, verify the validity of the estimate using a variety of robustness checks.\n\nThis workflow can be captured by four key verbs in DoWhy:\n\n- model\n- identify\n- estimate\n- refute\n\nUsing these verbs, DoWhy implements a causal inference engine that can support \na variety of methods. *model* encodes prior knowledge as a formal causal graph, *identify* uses \ngraph-based methods to identify the causal effect, *estimate* uses  \nstatistical methods for estimating the identified estimand, and finally *refute* \ntries to refute the obtained estimate by testing robustness to assumptions.\n\nKey differences compared to available causal inference software\n----------------------------------------------------------------\nDoWhy brings three key differences compared to available software for causal inference:\n\n**Explicit identifying assumptions**\n    Assumptions are first-class citizens in DoWhy.\n\n    Each analysis starts with a\n    building a causal model. The assumptions can be viewed graphically or in terms\n    of conditional independence statements. Wherever possible, DoWhy can also\n    automatically test for stated assumptions using observed data.\n\n**Separation between identification and estimation**\n    Identification is the causal problem. Estimation is simply a statistical problem.\n\n    DoWhy\n    respects this boundary and treats them separately. This focuses the causal\n    inference effort on identification, and frees up estimation using any\n    available statistical estimator for a target estimand. In addition, multiple\n    estimation methods can be used for a single identified_estimand and\n    vice-versa.\n\n**Automated robustness checks**\n    What happens when key identifying assumptions may not be satisfied?\n\n    The most critical, and often skipped, part of causal analysis is checking the\n    robustness of an estimate to unverified assumptions. DoWhy makes it easy to\n    automatically run sensitivity and robustness checks on the obtained estimate.\n\nFinally, DoWhy is easily extensible, allowing other implementations of the\nfour verbs to co-exist (e.g., we support implementations of the *estimation* verb from \nEconML and CausalML libraries). The four verbs are mutually independent, so their\nimplementations can be combined in any way.\n\n\n\nBelow are more details about the current implementation of each of these verbs.\n\nFour steps of causal inference\n===============================\n\nI. Model a causal problem\n-----------------------------\n\nDoWhy creates an underlying causal graphical model for each problem. This\nserves to make each causal assumption explicit. This graph need not be\ncomplete---you can provide a partial graph, representing prior\nknowledge about some of the variables. DoWhy automatically considers the rest\nof the variables as potential confounders.\n\nCurrently, DoWhy supports two formats for graph input: `gml <https://github.com/GunterMueller/UNI_PASSAU_FMI_Graph_Drawing>`_ (preferred) and\n`dot <http://www.graphviz.org/documentation/>`_. We strongly suggest to use gml as the input format, as it works well with networkx. You can provide the graph either as a .gml file or as a string. If you prefer to use dot format, you will need to install additional packages (pydot or pygraphviz, see the installation section above). Both .dot files and string format are supported. \n\nWhile not recommended, you can also specify common causes and/or instruments directly\ninstead of providing a graph.\n\nSupported formats for specifying causal assumptions\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n\n* **Graph**: Provide a causal graph in either gml or dot format. Can be a text file\n  or a string.\n* **Named variable sets**: Instead of the graph, provide variable names that\n  correspond to relevant categories, such as common causes, instrumental variables, effect\n  modifiers, frontdoor variables, etc.\n\nExamples of how to instantiate a causal model are in the `Getting Started\n<https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_simple_example.ipynb>`_\nnotebook.\n\n.. i comment image:: causal_model.png\n\nII. Identify a target estimand under the model\n----------------------------------------------\n\nBased on the causal graph, DoWhy finds all possible ways of identifying a desired causal effect based on\nthe graphical model. It uses graph-based criteria and do-calculus to find\npotential ways find expressions that can identify the causal effect. \n\nSupported identification criteria\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n\n* Back-door criterion\n* Front-door criterion\n* Instrumental Variables\n* Mediation (Direct and indirect effect identification)\n\nDifferent notebooks illustrate how to use these identification criteria. Check\nout the `Simple Backdoor <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_confounder_example.ipynb>`_ notebook for the back-door criterion, and the `Simple IV <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy-simple-iv-example.ipynb>`_ notebook for the instrumental variable criterion.\n\nIII. Estimate causal effect based on the identified estimand\n------------------------------------------------------------\n\nDoWhy supports methods based on both back-door criterion and instrumental\nvariables. It also provides a non-parametric confidence intervals and a permutation test for testing\nthe statistical significance of obtained estimate. \n\nSupported estimation methods \n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n\n* Methods based on estimating the treatment assignment\n    * Propensity-based Stratification\n    * Propensity Score Matching\n    * Inverse Propensity Weighting\n\n* Methods based on estimating the outcome model\n    * Linear Regression\n    * Generalized Linear Models\n\n* Methods based on the instrumental variable equation\n    * Binary Instrument/Wald Estimator\n    * Two-stage least squares\n    * Regression discontinuity\n\n* Methods for front-door criterion and general mediation\n    * Two-stage linear regression\n\nExamples of using these methods are in the `Estimation methods\n<https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_estimation_methods.ipynb>`_\nnotebook. \n\nUsing EconML and CausalML estimation methods in DoWhy\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\nIt is easy to call external estimation methods using DoWhy. Currently we\nsupport integrations with the `EconML <https://github.com/microsoft/econml>`_ and `CausalML <https://github.com/uber/causalml>`_ packages. Here\'s an example\nof estimating conditional treatment effects using EconML\'s double machine\nlearning estimator.\n\n.. code:: python\n\t\n\tfrom sklearn.preprocessing import PolynomialFeatures\n\tfrom sklearn.linear_model import LassoCV\n\tfrom sklearn.ensemble import GradientBoostingRegressor\n\tdml_estimate = model.estimate_effect(identified_estimand, method_name="backdoor.econml.dml.DML",\n                        control_value = 0,\n                        treatment_value = 1,\n                        target_units = lambda df: df["X0"]>1,\n                        confidence_intervals=False,\n                        method_params={\n                            "init_params":{\'model_y\':GradientBoostingRegressor(),\n                                           \'model_t\': GradientBoostingRegressor(),\n                                           \'model_final\':LassoCV(), \n                                           \'featurizer\':PolynomialFeatures(degree=1, include_bias=True)},\n                            "fit_params":{}}\n\t\t\t\t\t\t)\n\n\nMore examples are in the `Conditional Treatment Effects with DoWhy\n<https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy-conditional-treatment-effects.ipynb>`_ notebook. \n\nIV. Refute the obtained estimate\n-------------------------------------\nHaving access to multiple refutation methods to validate an effect estimate from a\ncausal estimator is\na key benefit of using DoWhy.\n\nSupported refutation methods\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n\n* **Add Random Common Cause**: Does the estimation method change its estimate after\n  we add an independent random variable as a common cause to the dataset?\n  (*Hint: It should not*)\n* **Placebo Treatment**: What happens to the estimated causal effect when we\n  replace the true treatment variable with an independent random variable?\n  (*Hint: the effect should go to zero*)\n* **Dummy Outcome**: What happens to the estimated causal effect when we replace\n  the true outcome variable with an independent random variable? (*Hint: The\n  effect should go to zero*)\n* **Simulated Outcome**: What happens to the estimated causal effect when we\n  replace the dataset with a simulated dataset based on a known data-generating\n  process closest to the given dataset? (*Hint: It should match the effect parameter\n  from the data-generating process*)\n* **Add Unobserved Common Causes**: How sensitive is the effect estimate when we\n  add an additional common cause (confounder) to the dataset that is correlated\n  with the treatment and the outcome? (*Hint: It should not be too sensitive*)\n* **Data Subsets Validation**: Does the estimated effect change significantly when\n  we replace the given dataset with a randomly selected subset? (*Hint: It\n  should not*)\n* **Bootstrap Validation**: Does the estimated effect change significantly when we \n  replace the given dataset with bootstrapped samples from the same dataset? (*Hint: It should not*)\n\nExamples of using refutation methods are in the `Refutations <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_refuter_notebook.ipynb>`_ notebook. For an advanced refutation that uses a simulated dataset based on user-provided or learnt data-generating processes, check out the `Dummy Outcome Refuter <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_demo_dummy_outcome_refuter.ipynb>`_ notebook. \nAs a practical example, `this notebook <https://github.com/microsoft/dowhy/blob/master/docs/source/example_notebooks/dowhy_refutation_testing.ipynb>`_ shows an application of refutation methods on evaluating effect estimators for the Infant Health and Development Program (IHDP) and Lalonde datasets. \n\nCiting this package\n====================\nIf you find DoWhy useful for your research work, please cite us as follows:\n\nAmit Sharma, Emre Kiciman, et al. DoWhy: A Python package for causal inference. 2019. https://github.com/microsoft/dowhy\n\nBibtex::\n\n  @misc{dowhy,\n  author={Sharma, Amit and Kiciman, Emre and others},\n  title={Do{W}hy: {A Python package for causal inference}},\n  howpublished={https://github.com/microsoft/dowhy},\n  year={2019}\n  }\n\nAlternatively, you can cite our Arxiv paper on DoWhy. \n\nAmit Sharma, Emre Kiciman. DoWhy: An End-to-End Library for Causal Inference. 2020. https://arxiv.org/abs/2011.04216\n\nBibtex::\n\n  @article{dowhypaper,\n  title={DoWhy: An End-to-End Library for Causal Inference},\n  author={Sharma, Amit and Kiciman, Emre},\n  journal={arXiv preprint arXiv:2011.04216},\n  year={2020}\n  }\n\nAnd if you find the gcm package useful for your work, please also cite us as:\n\nPatrick Blöbaum, Peter Götz, Kailash Budhathoki, Atalanti A. Mastakouri, Dominik Janzing. DoWhy-GCM: An extension of DoWhy for causal inference in graphical causal models. 2022. https://arxiv.org/abs/2206.06821\n\nBibtex::\n\n    @article{dowhy_gcm,\n      author = {Bl{\\"o}baum, Patrick and G{\\"o}tz, Peter and Budhathoki, Kailash and Mastakouri, Atalanti A. and Janzing, Dominik},\n      title = {DoWhy-GCM: An extension of DoWhy for causal inference in graphical causal models},\n      journal={arXiv preprint arXiv:2206.06821},\n      year={2022}\n    }\n\nRoadmap \n=======\nThe `projects <https://github.com/microsoft/dowhy/projects>`_ page lists the next steps for DoWhy. If you would like to contribute, have a look at the current projects. If you have a specific request for DoWhy, please `raise an issue <https://github.com/microsoft/dowhy/issues>`_.\n\nContributing\n============\n\nThis project welcomes contributions and suggestions. For a guide to contributing and a list of all contributors, check out `CONTRIBUTING.md <https://github.com/microsoft/dowhy/blob/master/CONTRIBUTING.md>`_. Our contributor code of conduct is available [here](https://github.com/py-why/governance/blob/main/CODE-OF-CONDUCT.md). You can also join the DoWhy development channel on Discord: |discord|_\n\n.. |discord| image:: https://img.shields.io/discord/818456847551168542\n.. _discord: https://discord.gg/cSBGb3vsZb\n\n',
    'author': 'PyWhy Community',
    'author_email': None,
    'maintainer': None,
    'maintainer_email': None,
    'url': 'https://github.com/pywhy/dowhy',
    'packages': packages,
    'package_data': package_data,
    'install_requires': install_requires,
    'extras_require': extras_require,
    'python_requires': '>=3.8,<3.11',
}


setup(**setup_kwargs)

```

package contents:

```
dowhy-0.8.0
dowhy-0.8.0/dowhy
dowhy-0.8.0/dowhy/data_transformers
dowhy-0.8.0/dowhy/data_transformers/pca_reducer.py
dowhy-0.8.0/dowhy/data_transformers/__init__.py
dowhy-0.8.0/dowhy/gcm
dowhy-0.8.0/dowhy/gcm/confidence_intervals_cms.py
dowhy-0.8.0/dowhy/gcm/shapley.py
dowhy-0.8.0/dowhy/gcm/anomaly_scorer.py
dowhy-0.8.0/dowhy/gcm/independence_test
dowhy-0.8.0/dowhy/gcm/independence_test/kernel_operation.py
dowhy-0.8.0/dowhy/gcm/independence_test/regression.py
dowhy-0.8.0/dowhy/gcm/independence_test/kernel.py
dowhy-0.8.0/dowhy/gcm/independence_test/__init__.py
dowhy-0.8.0/dowhy/gcm/confidence_intervals.py
dowhy-0.8.0/dowhy/gcm/uncertainty.py
dowhy-0.8.0/dowhy/gcm/distribution_change.py
dowhy-0.8.0/dowhy/gcm/whatif.py
dowhy-0.8.0/dowhy/gcm/stats.py
dowhy-0.8.0/dowhy/gcm/anomaly.py
dowhy-0.8.0/dowhy/gcm/constant.py
dowhy-0.8.0/dowhy/gcm/graph.py
dowhy-0.8.0/dowhy/gcm/influence.py
dowhy-0.8.0/dowhy/gcm/validation.py
dowhy-0.8.0/dowhy/gcm/density_estimator.py
dowhy-0.8.0/dowhy/gcm/config.py
dowhy-0.8.0/dowhy/gcm/stochastic_models.py
dowhy-0.8.0/dowhy/gcm/ml
dowhy-0.8.0/dowhy/gcm/ml/classification.py
dowhy-0.8.0/dowhy/gcm/ml/regression.py
dowhy-0.8.0/dowhy/gcm/ml/__init__.py
dowhy-0.8.0/dowhy/gcm/density_estimators.py
dowhy-0.8.0/dowhy/gcm/divergence.py
dowhy-0.8.0/dowhy/gcm/util
dowhy-0.8.0/dowhy/gcm/util/plotting.py
dowhy-0.8.0/dowhy/gcm/util/general.py
dowhy-0.8.0/dowhy/gcm/util/__init__.py
dowhy-0.8.0/dowhy/gcm/util/pygraphviz.py
dowhy-0.8.0/dowhy/gcm/fitting_sampling.py
dowhy-0.8.0/dowhy/gcm/cms.py
dowhy-0.8.0/dowhy/gcm/anomaly_scorers.py
dowhy-0.8.0/dowhy/gcm/auto.py
dowhy-0.8.0/dowhy/gcm/__init__.py
dowhy-0.8.0/dowhy/gcm/_noise.py
dowhy-0.8.0/dowhy/gcm/feature.py
dowhy-0.8.0/dowhy/gcm/fcms.py
dowhy-0.8.0/dowhy/causal_identifier.py
dowhy-0.8.0/dowhy/causal_model.py
dowhy-0.8.0/dowhy/plotter.py
dowhy-0.8.0/dowhy/causal_graph.py
dowhy-0.8.0/dowhy/VERSION
dowhy-0.8.0/dowhy/causal_identifiers
dowhy-0.8.0/dowhy/causal_identifiers/id_identifier.py
dowhy-0.8.0/dowhy/causal_identifiers/backdoor.py
dowhy-0.8.0/dowhy/causal_identifiers/__init__.py
dowhy-0.8.0/dowhy/causal_refuter.py
dowhy-0.8.0/dowhy/datasets.py
dowhy-0.8.0/dowhy/do_sampler.py
dowhy-0.8.0/dowhy/causal_estimator.py
dowhy-0.8.0/dowhy/graph_learner.py
dowhy-0.8.0/dowhy/data_transformer.py
dowhy-0.8.0/dowhy/interpreters
dowhy-0.8.0/dowhy/interpreters/confounder_distribution_interpreter.py
dowhy-0.8.0/dowhy/interpreters/textual_interpreter.py
dowhy-0.8.0/dowhy/interpreters/propensity_balance_interpreter.py
dowhy-0.8.0/dowhy/interpreters/__init__.py
dowhy-0.8.0/dowhy/interpreters/visual_interpreter.py
dowhy-0.8.0/dowhy/interpreters/textual_effect_interpreter.py
dowhy-0.8.0/dowhy/causal_refuters
dowhy-0.8.0/dowhy/causal_refuters/graph_refuter.py
dowhy-0.8.0/dowhy/causal_refuters/random_common_cause.py
dowhy-0.8.0/dowhy/causal_refuters/data_subset_refuter.py
dowhy-0.8.0/dowhy/causal_refuters/bootstrap_refuter.py
dowhy-0.8.0/dowhy/causal_refuters/placebo_treatment_refuter.py
dowhy-0.8.0/dowhy/causal_refuters/add_unobserved_common_cause.py
dowhy-0.8.0/dowhy/causal_refuters/linear_sensitivity_analyzer.py
dowhy-0.8.0/dowhy/causal_refuters/__init__.py
dowhy-0.8.0/dowhy/causal_refuters/dummy_outcome_refuter.py
dowhy-0.8.0/dowhy/utils
dowhy-0.8.0/dowhy/utils/cli_helpers.py
dowhy-0.8.0/dowhy/utils/propensity_score.py
dowhy-0.8.0/dowhy/utils/api.py
dowhy-0.8.0/dowhy/utils/dgp.py
dowhy-0.8.0/dowhy/utils/cit.py
dowhy-0.8.0/dowhy/utils/ordered_set.py
dowhy-0.8.0/dowhy/utils/graph_operations.py
dowhy-0.8.0/dowhy/utils/__init__.py
dowhy-0.8.0/dowhy/utils/dgps
dowhy-0.8.0/dowhy/utils/dgps/linear_dgp.py
dowhy-0.8.0/dowhy/utils/dgps/random_neural_network.py
dowhy-0.8.0/dowhy/utils/dgps/cubic_dgp.py
dowhy-0.8.0/dowhy/utils/dgps/quadratic_dgp.py
dowhy-0.8.0/dowhy/causal_estimators
dowhy-0.8.0/dowhy/causal_estimators/linear_regression_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/propensity_score_stratification_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/generalized_linear_model_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/propensity_score_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/econml.py
dowhy-0.8.0/dowhy/causal_estimators/two_stage_regression_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/regression_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/distance_matching_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/propensity_score_matching_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/instrumental_variable_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/propensity_score_weighting_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/regression_discontinuity_estimator.py
dowhy-0.8.0/dowhy/causal_estimators/causalml.py
dowhy-0.8.0/dowhy/causal_estimators/__init__.py
dowhy-0.8.0/dowhy/interpreter.py
dowhy-0.8.0/dowhy/do_samplers
dowhy-0.8.0/dowhy/do_samplers/kernel_density_sampler.py
dowhy-0.8.0/dowhy/do_samplers/multivariate_weighting_sampler.py
dowhy-0.8.0/dowhy/do_samplers/weighting_sampler.py
dowhy-0.8.0/dowhy/do_samplers/mcmc_sampler.py
dowhy-0.8.0/dowhy/do_samplers/__init__.py
dowhy-0.8.0/dowhy/graph_learners
dowhy-0.8.0/dowhy/graph_learners/cdt.py
dowhy-0.8.0/dowhy/graph_learners/ges.py
dowhy-0.8.0/dowhy/graph_learners/lingam.py
dowhy-0.8.0/dowhy/graph_learners/__init__.py
dowhy-0.8.0/dowhy/__init__.py
dowhy-0.8.0/dowhy/api
dowhy-0.8.0/dowhy/api/causal_data_frame.py
dowhy-0.8.0/dowhy/api/__init__.py
dowhy-0.8.0/tests
dowhy-0.8.0/tests/data_transformers
dowhy-0.8.0/tests/data_transformers/test_pca_reducer.py
dowhy-0.8.0/tests/gcm
dowhy-0.8.0/tests/gcm/test_graph.py
dowhy-0.8.0/tests/gcm/test_density_estimators.py
dowhy-0.8.0/tests/gcm/independence_test
dowhy-0.8.0/tests/gcm/independence_test/test_kernel.py
dowhy-0.8.0/tests/gcm/independence_test/test_regression.py
dowhy-0.8.0/tests/gcm/independence_test/__init__.py
dowhy-0.8.0/tests/gcm/test_feature.py
dowhy-0.8.0/tests/gcm/test_validation.py
dowhy-0.8.0/tests/gcm/test_confidence_intervals_cms.py
dowhy-0.8.0/tests/gcm/test_anomaly_attribution.py
dowhy-0.8.0/tests/gcm/test_arrow_strength.py
dowhy-0.8.0/tests/gcm/test_ml.py
dowhy-0.8.0/tests/gcm/test_distribution_change.py
dowhy-0.8.0/tests/gcm/test_anomaly.py
dowhy-0.8.0/tests/gcm/test_shapley.py
dowhy-0.8.0/tests/gcm/test_intrinsic_influence.py
dowhy-0.8.0/tests/gcm/util
dowhy-0.8.0/tests/gcm/util/test_pygraphviz.py
dowhy-0.8.0/tests/gcm/util/__init__.py
dowhy-0.8.0/tests/gcm/util/test_plotting.py
dowhy-0.8.0/tests/gcm/test_stochastic_models.py
dowhy-0.8.0/tests/gcm/test_confidence_intervals.py
dowhy-0.8.0/tests/gcm/test_divergence.py
dowhy-0.8.0/tests/gcm/test_auto.py
dowhy-0.8.0/tests/gcm/test_anomaly_scorers.py
dowhy-0.8.0/tests/gcm/__init__.py
dowhy-0.8.0/tests/gcm/test_uncertainty.py
dowhy-0.8.0/tests/gcm/test_whatif.py
dowhy-0.8.0/tests/gcm/test_fcms.py
dowhy-0.8.0/tests/gcm/test_stats.py
dowhy-0.8.0/tests/sample_dag.txt
dowhy-0.8.0/tests/causal_identifiers
dowhy-0.8.0/tests/causal_identifiers/test_id_identifier.py
dowhy-0.8.0/tests/causal_identifiers/example_graphs.py
dowhy-0.8.0/tests/causal_identifiers/base.py
dowhy-0.8.0/tests/causal_identifiers/test_optimize_backdoor.py
dowhy-0.8.0/tests/causal_identifiers/__init__.py
dowhy-0.8.0/tests/causal_identifiers/test_backdoor_identifier.py
dowhy-0.8.0/tests/do_sampler
dowhy-0.8.0/tests/do_sampler/test_pandas_do_api.py
dowhy-0.8.0/tests/test_causal_estimator.py
dowhy-0.8.0/tests/test_causal_refuter.py
dowhy-0.8.0/tests/definitions.py
dowhy-0.8.0/tests/causal_refuters
dowhy-0.8.0/tests/causal_refuters/test_dummy_outcome_refuter.py
dowhy-0.8.0/tests/causal_refuters/test_bootstrap_refuter.py
dowhy-0.8.0/tests/causal_refuters/test_placebo_refuter.py
dowhy-0.8.0/tests/causal_refuters/test_data_subset_refuter.py
dowhy-0.8.0/tests/causal_refuters/test_add_unobserved_common_cause.py
dowhy-0.8.0/tests/causal_refuters/base.py
dowhy-0.8.0/tests/causal_refuters/__init__.py
dowhy-0.8.0/tests/conftest.py
dowhy-0.8.0/tests/test_causal_model.py
dowhy-0.8.0/tests/causal_estimators
dowhy-0.8.0/tests/causal_estimators/test_propensity_score_matching_estimator.py
dowhy-0.8.0/tests/causal_estimators/test_causalml_estimator.py
dowhy-0.8.0/tests/causal_estimators/test_instrumental_variable_estimator.py
dowhy-0.8.0/tests/causal_estimators/test_generalized_linear_model_estimator.py
dowhy-0.8.0/tests/causal_estimators/test_propensity_score_weighting_estimator.py
dowhy-0.8.0/tests/causal_estimators/test_propensity_score_stratification_estimator.py
dowhy-0.8.0/tests/causal_estimators/test_econml_estimator.py
dowhy-0.8.0/tests/causal_estimators/test_linear_regression_estimator.py
dowhy-0.8.0/tests/causal_estimators/test_two_stage_regression_estimator.py
dowhy-0.8.0/tests/causal_estimators/base.py
dowhy-0.8.0/tests/causal_estimators/test_regression_discontinuity_estimator.py
dowhy-0.8.0/tests/causal_estimators/mock_external_estimator.py
dowhy-0.8.0/tests/causal_estimators/__init__.py
dowhy-0.8.0/tests/__init__.py
dowhy-0.8.0/tests/test_notebooks.py
dowhy-0.8.0/tests/test_data_transformer.py
dowhy-0.8.0/LICENSE
dowhy-0.8.0/pyproject.toml
dowhy-0.8.0/docs
dowhy-0.8.0/docs/source
dowhy-0.8.0/docs/source/dowhy.do_samplers.rst
dowhy-0.8.0/docs/source/user_guide
dowhy-0.8.0/docs/source/user_guide/causality_intro.rst
dowhy-0.8.0/docs/source/user_guide/effect_inference
dowhy-0.8.0/docs/source/user_guide/effect_inference/model.rst
dowhy-0.8.0/docs/source/user_guide/effect_inference/refute.rst
dowhy-0.8.0/docs/source/user_guide/effect_inference/index.rst
dowhy-0.8.0/docs/source/user_guide/effect_inference/estimate.rst
dowhy-0.8.0/docs/source/user_guide/effect_inference/identify.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/scm.png
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions/index.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions/quantify_arrow_strength.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions/simulate_impact_of_interventions.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions/attribute_distributional_changes.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions/quantify_intrinsic_causal_influence.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions/average_causal_effect.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions/arrow_strength_example.png
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/answering_causal_questions/computing_counterfactuals.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/pcm.png
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/estimating_confidence_intervals.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/index.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/customizing_model_assignment.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/draw_samples.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/introduction.rst
dowhy-0.8.0/docs/source/user_guide/gcm_based_inference/independence_tests.rst
dowhy-0.8.0/docs/source/dowhy.rst
dowhy-0.8.0/docs/source/getting_started
dowhy-0.8.0/docs/source/getting_started/intro.rst
dowhy-0.8.0/docs/source/getting_started/cite.rst
dowhy-0.8.0/docs/source/getting_started/comparison.rst
dowhy-0.8.0/docs/source/getting_started/install.rst
dowhy-0.8.0/docs/source/dowhy.api.rst
dowhy-0.8.0/docs/source/_static
dowhy-0.8.0/docs/source/_static/regression_output.png
dowhy-0.8.0/docs/source/contributing
dowhy-0.8.0/docs/source/contributing/index.rst
dowhy-0.8.0/docs/source/dowhy.causal_estimators.rst
dowhy-0.8.0/docs/source/index.rst
dowhy-0.8.0/docs/source/example_notebooks
dowhy-0.8.0/docs/source/example_notebooks/graph_conditional_independence_refuter.ipynb
dowhy-0.8.0/docs/source/example_notebooks/supply_chain_week_over_week.csv
dowhy-0.8.0/docs/source/example_notebooks/rca_microservice_architecture_anomaly_1000.csv
dowhy-0.8.0/docs/source/example_notebooks/dowhy_simple_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_optimize_backdoor_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/nb_casestudies_index.rst
dowhy-0.8.0/docs/source/example_notebooks/dowhy-conditional-treatment-effects.ipynb
dowhy-0.8.0/docs/source/example_notebooks/gcm_supply_chain_dist_change.ipynb
dowhy-0.8.0/docs/source/example_notebooks/gcm_rca_microservice_architecture.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_ihdp_data_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/patients_database.csv
dowhy-0.8.0/docs/source/example_notebooks/rca_microservice_architecture_anomaly.csv
dowhy-0.8.0/docs/source/example_notebooks/gcm_401k_analysis.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_refuter_notebook.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_estimation_methods.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_example_effect_of_memberrewards_program.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_causal_discovery_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/microservice-architecture-dependencies.png
dowhy-0.8.0/docs/source/example_notebooks/load_graph_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/rca_microservice_architecture_latencies.csv
dowhy-0.8.0/docs/source/example_notebooks/dowhy_demo_dummy_outcome_refuter.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_lalonde_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/sensitivity_analysis_testing.ipynb
dowhy-0.8.0/docs/source/example_notebooks/shifting-resources.png
dowhy-0.8.0/docs/source/example_notebooks/dowhy_causal_api.ipynb
dowhy-0.8.0/docs/source/example_notebooks/nb_index.rst
dowhy-0.8.0/docs/source/example_notebooks/do_sampler_demo.ipynb
dowhy-0.8.0/docs/source/example_notebooks/gcm_counterfactual_medical_dry_eyes.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_confounder_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/pension.csv
dowhy-0.8.0/docs/source/example_notebooks/newly_come_patients.csv
dowhy-0.8.0/docs/source/example_notebooks/dowhy_mediation_analysis.ipynb
dowhy-0.8.0/docs/source/example_notebooks/identifying_effects_using_id_algorithm.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_refutation_testing.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_interpreter.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_ranking_methods.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy-simple-iv-example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/gcm_draw_samples.ipynb
dowhy-0.8.0/docs/source/example_notebooks/images
dowhy-0.8.0/docs/source/example_notebooks/images/supply-chain.png
dowhy-0.8.0/docs/source/example_notebooks/images/causality_ml_example_challenges.png
dowhy-0.8.0/docs/source/example_notebooks/images/causalinference_schematic.png
dowhy-0.8.0/docs/source/example_notebooks/images/supervised_ml_schematic.png
dowhy-0.8.0/docs/source/example_notebooks/images/real_vs_counterfactual_world.png
dowhy-0.8.0/docs/source/example_notebooks/tutorial-causalinference-machinelearning-using-dowhy-econml.ipynb
dowhy-0.8.0/docs/source/example_notebooks/gcm_basic_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/lalonde_pandas_api.ipynb
dowhy-0.8.0/docs/source/example_notebooks/DoWhy-The Causal Story Behind Hotel Booking Cancellations.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_twins_example.ipynb
dowhy-0.8.0/docs/source/example_notebooks/dowhy_multiple_treatments.ipynb
dowhy-0.8.0/docs/source/modules.rst
dowhy-0.8.0/docs/source/dowhy.causal_refuters.rst
dowhy-0.8.0/docs/source/dowhy.gcm.ml.rst
dowhy-0.8.0/docs/source/dowhy.causal_identifiers.rst
dowhy-0.8.0/docs/source/dowhy.data_transformers.rst
dowhy-0.8.0/docs/source/dowhy.interpreters.rst
dowhy-0.8.0/docs/source/dowhy.graph_learners.rst
dowhy-0.8.0/docs/source/dowhy.gcm.util.rst
dowhy-0.8.0/docs/source/dowhy.gcm.independence_test.rst
dowhy-0.8.0/docs/source/conf.py
dowhy-0.8.0/docs/source/dowhy.gcm.rst
dowhy-0.8.0/docs/source/code_repo.rst
dowhy-0.8.0/docs/source/Makefile
dowhy-0.8.0/docs/source/example_graphs
dowhy-0.8.0/docs/source/example_graphs/simple_graph_example.dot
dowhy-0.8.0/docs/source/example_graphs/simple_graph_example.gml
dowhy-0.8.0/docs/source/dowhy.utils.rst
dowhy-0.8.0/docs/update_docs.sh
dowhy-0.8.0/docs/Makefile
dowhy-0.8.0/docs/images
dowhy-0.8.0/docs/images/regression_output.png
dowhy-0.8.0/docs/images/do_barplot.png
dowhy-0.8.0/docs/images/dowhy-schematic.png
dowhy-0.8.0/setup.py
dowhy-0.8.0/CONTRIBUTING.md
dowhy-0.8.0/PKG-INFO
dowhy-0.8.0/README.rst
```